File: webapp/src/main/java/com/miaskor/todo_spring/connectors/rest/api/ClientConnector.java
package com.miaskor.todo_spring.connectors.rest.api;

import by.miaskor.domain.rest.api.ClientRequestDto;
import by.miaskor.domain.rest.api.ClientResponseDto;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;

@FeignClient(value = "rest-api-client", url = "${connector.rest-api.url}")
public interface ClientConnector {

  @RequestMapping(value = "/clients/auth", method = RequestMethod.GET)
  ClientResponseDto getClientByLoginAndPassword(@RequestParam String login, @RequestParam String password);

  @RequestMapping(value = "/clients", method = RequestMethod.POST)
  ClientResponseDto createClient(@RequestBody ClientRequestDto clientRequestDto);
}

File: rest-api/rest-connector/src/main/kotlin/by/miaskor/rest/connector/ClientConnector.kt
package by.miaskor.rest.connector

import by.miaskor.rest.dto.ClientDtoRequest
import by.miaskor.rest.dto.ClientDtoResponse
import feign.Headers
import feign.Param
import feign.RequestLine

interface ClientConnector {

  @RequestLine("GET /auth?login={login}&password={password}")
  fun getClientByLoginAndPassword(@Param("login") login: String, @Param("password") password: String): ClientDtoResponse

  @RequestLine("POST")
  @Headers("Content-Type: application/json")
  fun createClient(clientDtoRequest: ClientDtoRequest): ClientDtoResponse
}
